src/coinify/trade.js

```diff
@@ -274,15 +277,28 @@ CoinifyTrade.buy = function (quote, medium, coinify) {
   var account = MyWallet.wallet.hdwallet.defaultAccount;
 
   var receiveAddressIndex = account.receiveIndex;
+  var baseLabel = 'Coinify order #';
 
   // Respect the GAP limit:
   if (receiveAddressIndex - account.lastUsedReceiveIndex >= 19) {
-    return Promise.reject('gap_limit');
+    var orderRegex = /^Coinify order (#\d+,\s)*(#\d+)$/;
+    for (var i = 1; i < 20; i++) {
```

Comment:
Similar to my previous remark: rather than looping through address labels (brittle), consider looping through coinify.trades.